comp/core/workloadmeta/collectors/util/image_util.go

```diff
@@ -32,18 +36,62 @@ func UpdateSBOMRepoMetadata(sbom *workloadmeta.SBOM, repoTags, repoDigests []str
 	}
 
 	properties := *sbom.CycloneDXBOM.Metadata.Component.Properties
-	properties = removeOldRepoProperties(properties)
+	mismatched := !propertiesEqualsValues(properties, repoTags, repoTagPropertyKey) || !propertiesEqualsValues(properties, repoDigests, repoDigestPropertyKey)
 
-	properties = appendRepoProperties(properties, repoTags, repoTagPropertyKey)
-	properties = appendRepoProperties(properties, repoDigests, repoDigestPropertyKey)
+	if mismatched {
+		sbom = deepcopy.Copy(sbom).(*workloadmeta.SBOM)
+		newProperties := *sbom.CycloneDXBOM.Metadata.Component.Properties
+		newProperties = cleanProperties(newProperties)
 
-	sbom.CycloneDXBOM.Metadata.Component.Properties = &properties
+		newProperties = appendProperties(newProperties, repoTags, repoTagPropertyKey)
+		newProperties = appendProperties(newProperties, repoDigests, repoDigestPropertyKey)
+
+		// Sort properties to ensure consistent ordering for tests
+		sort.Slice(newProperties, func(i, j int) bool {
+			return newProperties[i].Name < newProperties[j].Name || newProperties[i].Value < newProperties[j].Value
+		})
+		sbom.CycloneDXBOM.Metadata.Component.Properties = &newProperties
+	}
 
 	return sbom
 }
 
-// removeOldRepoProperties returns an array without repodigests and repoTags
-func removeOldRepoProperties(properties []cyclonedx.Property) []cyclonedx.Property {
+// propertiesEqualsValues function compares the existing properties with the new values
+func propertiesEqualsValues(properties []cyclonedx.Property, newValues []string, propertyKeyType string) bool {
+	existingValuesMap := make(map[string]struct{})
+	for _, prop := range properties {
+		if prop.Name == propertyKeyType {
+			existingValuesMap[prop.Value] = struct{}{}
+		}
+	}
+
+	for _, newValue := range newValues {
+		if _, found := existingValuesMap[newValue]; !found {
+			return false
+		}
+	}
+
+	for existingValue := range existingValuesMap {
+		if !contains(newValues, existingValue) {
+			return false
+		}
+	}
+
+	return true
+}
+
+// contains is a helper function to check if the slice contains the string value
+func contains(slice []string, value string) bool {
```

Comment:
do we have already a generic function for this purpose?

Reply:
I did a quick search in the code base and didn't find a util function named `Contains`. I guess it could already exist